setup.py

```diff
@@ -26,18 +26,84 @@
 import distutils.ccompiler
 from distutils.dep_util import newer
 from distutils.command.install import install as _install
-import sys, glob, os
+import sys, glob, os, fnmatch
 import subprocess
 import shlex
 
-import numpy as np
+import numpy
 
 from fofix.core import Version, SceneFactory
 
 from setuptools import setup, Extension, Command
 from Cython.Build import cythonize
 
 
+def glob_recursive(dir, file_pattern="*"):
+    """ Like glob, but recurses sub-dirs.
+    :param dir: directory from which to start searching
+    :param file_pattern: (default '*')
+        optional name pattern for desired files. Used to filter files at 
+        every level; NOT used to filter directories.
+    :return: list(str) of path to every file found.
+    """
+    matches = []
+    for subdir, dirnames, filenames in os.walk(dir):
+        for filename in fnmatch.filter(filenames, file_pattern):
+            matches.append(os.path.join(subdir, filename))
+    return matches
+
+def to_data_files(target, top, files):
+    """ Convert plain list of files to something data_files would like.
+    Each file's location will be computed relative to 'top'; the resulting sub-path
+    will be the location under 'target' where py2exe will place the file.
+    The result is effectively a set of directories inside of 'target'
+    that reproduce the structure of directories inside 'top'.
+    This lets you move a set of files from any location in your source tree
+    into any location in the distro.
+     
+    :param target: str - relative path inside of py2exe distro where files shall go
+    :param top: str - absolute (or relative to system current) path to remove from each of the paths in 'files'
+    :param files: list(str) of file paths to process
+    :return: list( (str, list(str)) ) - the structure required by py2exe.options.data_files
+    """
+    outs = {}
+    # normalize top
+    top = os.path.normpath(os.path.abspath(top))
+    for f in files:
+        # normalize each file
+        f = os.path.normpath(os.path.abspath(f))
+        # compute path relative to 'top'
+        rel = os.path.relpath(os.path.dirname(f), top)
+        subtarg = os.path.join(target, rel)
+        # save file under corresponding relative path
+        lst = outs.get(subtarg)
+        if not lst:
+            lst = list()
+            outs[subtarg] = lst
+        lst.append(f)
+    return [(k,outs[k]) for k in sorted(outs.keys())]
+
+
+# Hack to fix "No such file or directory: 'numpy-atlas.dll'".
+# https://stackoverflow.com/questions/36191770/py2exe-errno-2-no-such-file-or-directory-numpy-atlas-dll
+if os.name == 'nt':
```

Comment:
You can put this block within the block [L111](#diff-2eeaed663bd0d25b7e608891384b7298R111).

Reply:
Technically true, but it's not good organization to hide a hack inside the normal program flow. Also, that hack needs to execute before any of the normal setup-building begins.